pkg/rpctype/rpctype.go

```diff
@@ -72,6 +77,20 @@ type PollRes struct {
 	MaxSignal  signal.Serial
 }
 
+type RunnerConnectArgs struct {
+	Pool, VM int
+}
+
+type NewExchangeArgs struct {
```

Comment:
The below have comments for each field. It'd be good to have them here, too.

Also, "NewExchange" might not be the best name for a function/RPC. While "Exchange" is a verb, I'm not sure what's "New" about it, so a subject is missing. Probably "Prog" being the missing subject.

So my suggestions would be one of:
- GetNewProg{Args,Res}
- GetNextProg{Args,Res}
- ExchangeProg{Args,Res}
- ExchangeNewProg{Args,Res}
- ExchangeNextProg{Args,Res}

I'd actually go with "Next" and not "New", because it might be an old recycled program.

Pick one that sounds most reasonable.

Reply:
> The below have comments for each field. It'd be good to have them here, too.

Sure, wasn't sure if I should add comments in this file since most of the structs don't have comments. But if you think it's better, I will.

>  Also, "NewExchange" might not be the best name for a function/RPC. While "Exchange" is a verb, I'm not sure what's "New" about it, so a subject is missing. Probably "Prog" being the missing subject
...

While I agree that `NextExchange` might be a better option than `NewExchange`, I would still omit `Prog` from the name because it might lead the user/contributor to believe that the host and the guest are just exchanging programs when, in fact:
* the host passes a new program to the guest
* the guest might pass results to the host from an older program that has finished executed since the last `NextExchange` RPC call

Hope that makes sense.

Reply:
Typically a good name for a function has the format `<optional-namespace-name> + <verb> + <optional-adverb> + <subject>` (sometimes the subject can be implied, then it can be omitted). What you have here is `<adverb or subject> + <subject or verb>` because `Next` can be either an adverb or subject, and `Exchange` can be either a subject or verb.

My problem is mainly that the concatenation of `Next`/`New` and `Exchange` appears random because there is no subject. As is, `NewExchange` or `NextExchange` tells me nothing about what this RPC is supposed to do.

I guess the question is "exchange what? what is next?"

Going for full ambiguity or omission because the function does too much is not an excuse to try harder. Your 2 points suggest that we have "program" and "program results". So why not:
- `ExchangeResultAndProg`
- `SendResultAndGetNextProg`
- ... or some variation.

Long names are the unfortunate side-effect of having one function that does a lot.

While I'm not a fan of bikeshedding names, the problem here is this seems to be syzkaller-global and not restricted to the verf package, so it makes sense to get this right.

But in the end, if I'm the only one that wants more descriptive names, then I won't argue much further. ;-)

Reply:
I think you are right and `New(xt)Exchange` might not be a fully clear name (although I tried to explain in the function documentation what this "exchange" means). At the same time, in the design phase we decided on getting results and sending new programs via the same RPC call (especially if we are going to introduce batching and parallelisation at some point). Do you think going for a long, very explicit name would be better or should we think about another word rather than "exchange"? @dvyukov any opinions on this?

Reply:
I don't have great name candidates, and I don't have a strong opinion for your candidates.
I am thinking of something like "sync" or "poll" verbs, but I can't build complete name that is better than yours.
Let's go with some name and maybe just add a comment explaining what it's doing.

Reply:
I will add a comment for `NextExchangeArgs` and then we can change the name with a better alternative in the follow-up PR. Would that be fine? @dvyukov @melver